pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -445,16 +468,36 @@ func (nc *NodeController) PopulateAddresses(nodes *api.NodeList) (*api.NodeList,
 	return nodes, nil
 }
 
-func (nc *NodeController) recordNodeOfflineEvent(node *api.Node) {
-	ref := &api.ObjectReference{
-		Kind:      "Node",
-		Name:      node.Name,
-		UID:       types.UID(node.Name),
-		Namespace: "",
+func (nc *NodeController) getNodeReference(node *api.Node) *api.ObjectReference {
+	return &api.ObjectReference{
+		Kind: "Node",
+		Name: node.Name,
+		UID:  types.UID(node.Name),
 	}
+}
+
+func (nc *NodeController) recordNodeOnlineEvent(node *api.Node) {
```

Comment:
Thanks for adding these additional functions. It was kinda bad that we previously only had offline. But can you please add calls into these functions from MonitorNodeStatus()?